app/src/main/java/com/infomaniak/mail/ui/main/NoAnimSlidingPaneLayout.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Infomaniak Mail - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main
+
+import android.content.Context
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import androidx.slidingpanelayout.widget.SlidingPaneLayout
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+
+class NoAnimSlidingPaneLayout(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyle: Int,
+) : SlidingPaneLayout(context, attrs, defStyle) {
+
+    private var slideOffsetField: Field? = null
+    private var slideableViewField: Field? = null
+    private var preservedOpenStateField: Field? = null
+
+    private var parallaxOtherViewsMethod: Method? = null
+    private var dispatchOnPanelOpenedMethod: Method? = null
+    private var dispatchOnPanelClosedMethod: Method? = null
+    private var updateObscuredViewsVisibilityMethod: Method? = null
+
+    constructor(context: Context, attrs: AttributeSet? = null) : this(context, attrs, defStyle = 0)
```

Comment:
You could use `@JvmOverloads` to get all the constructors automatically instead of creating each different one manually

``` kt
class NoAnimSlidingPaneLayout @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0,
) : SlidingPaneLayout(context, attrs, defStyle) {
```